controllers/sample_controller_rendered_resources.go

```diff
@@ -0,0 +1,359 @@
+			logger.Error(err, "error during installation of resources")
+			r.Event(objectInstance, "Warning", "ResourcesInstall", "installing resources error")
+			return r.setStatusForObjectInstance(ctx, objectInstance, status.WithState(v1alpha1.StateError))
+		}
+	}
+
+	status.Conditions[0].Status = metav1.ConditionTrue
+	status.Conditions[0].ObservedGeneration = objectInstance.GetGeneration()
+	meta.SetStatusCondition(&status.Conditions, status.Conditions[0])
+	return r.setStatusForObjectInstance(ctx, objectInstance, status.WithState(v1alpha1.StateReady))
+}
+
+// HandleDeletingState processed the deletion on the reconciled resource.
+// Once the deletion if processed the relevant finalizers (if applied) are removed.
+func (r *SampleReconciler) HandleDeletingState(ctx context.Context, objectInstance *v1alpha1.Sample) error {
+	r.Event(objectInstance, "Warning", "Deleting", "resource deleting")
+	logger := log.FromContext(ctx)
+
+	status := getStatusFromObjectInstance(objectInstance)
+
+	resourceObjs, err := getResourcesFromLocalPath(objectInstance.Spec.ResourceFilePath, logger)
+	if err != nil {
+		return err
+	}
+	r.Event(objectInstance, "Normal", "ResourcesDelete", "deleting resources")
+
+	// the resources to be installed are unstructured,
+	// so please make sure the types are available on the target cluster
+	for _, obj := range resourceObjs.Items {
+		if err = r.Client.Delete(ctx, obj); err != nil && !errors2.IsNotFound(err) {
+			logger.Error(err, "error during installation of resources")
+			r.Event(objectInstance, "Warning", "ResourcesDelete", "deleting resources error")
+			return r.setStatusForObjectInstance(ctx, objectInstance, status.WithState(v1alpha1.StateError))
+		}
+	}
+
+	// if resources are ready to be deleted, remove finalizer
+	if controllerutil.RemoveFinalizer(objectInstance, finalizer) {
+		return r.Client.Update(ctx, objectInstance)
+	}
+	return nil
+}
+
+// HandleReadyState checks for the consistency of reconciled resource, by verifying the underlying resources.
+func (r *SampleReconciler) HandleReadyState(ctx context.Context, objectInstance *v1alpha1.Sample) error {
+	// TODO: handle custom ready state handling here
+	// by default we will call HandleProcessingState to verify if all resources were installed correctly
+	// by patching them again
+	return r.HandleProcessingState(ctx, objectInstance)
+}
+
+func (r *SampleReconciler) setStatusForObjectInstance(ctx context.Context, objectInstance *v1alpha1.Sample,
+	status v1alpha1.SampleStatus,
+) error {
+	objectInstance.Status = status
+
+	if err := r.ssaStatus(ctx, objectInstance); err != nil {
+		r.Event(objectInstance, "Warning", "ErrorUpdatingStatus", fmt.Sprintf("updating state to %v", string(status.State)))
+		return fmt.Errorf("error while updating status %s to: %w", status.State, err)
+	}
+
+	r.Event(objectInstance, "Normal", "StatusUpdated", fmt.Sprintf("updating state to %v", string(status.State)))
```

Comment:
I believe an event for every status update will spam the API server so I do not think we should recommend that.

Reply:
There will be status event update overload only when doing consistency checks, since it will call a status update on each consistency check. For this reason I added a TODO there to improve it. The second source could be on repeated errors. I will separate the logic so that duplicate status updates do not take place.